container/go/pkg/compat/reader.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2015 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//////////////////////////////////////////////////////////////////////
+// Reads an legacy image layout on disk.
+package compat
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/partial"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+	"github.com/google/go-containerregistry/pkg/v1/validate"
+	"github.com/pkg/errors"
+)
+
+// Read returns a docker image referenced by the legacy intermediate layout at src. The image index should have been outputted by container_pull.
+// NOTE: this only reads index with a single image.
+func Read(src string) (v1.Image, error) {
+	_, err := isValidLegacylayout(src)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Invalid legacy layout at %s, requires manifest.json, config.json and digest files.", src)
```

Comment:
error messages aren't standalone sentences. They should begin with a small letter and shouldn't end with a period.